### PR TITLE
[Quant] Add fused ConvAddReLU2d module for onednn backend

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91155
* __->__ #91154
* #91153
* #91152

**Summary**
Post op fusion can reduce data movement overhead and improve inference performance. This PR adds fused ConvAddReLU2d module for onednn backend, which will be used for int8 inference with onednn backend. Cannot call this module with other quantization backends otherwise an error is thrown.

**Test plan**
```
python -m pytest test_quantization.py -k test_conv2d_add_relu
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]
LGTM.

File: torch/ao/nn/intrinsic/quantized/modules/conv_add.py [hunk 1]
```diff
@@ -48,3 +48,46 @@ def from_float(cls, mod):
...
+    Attributes:
+        Same as torch.ao.nn.quantized.Conv2d
+
+    """
+    _FLOAT_MODULE = torch.ao.nn.intrinsic.ConvAddReLU2d  # type: ignore[assignment]
+
+    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
+                 padding=0, dilation=1, groups=1, bias=True,
+                 padding_mode='zeros', device=None, dtype=None):
+        super(ConvAddReLU2d, self).__init__(
```
Comment (jerryzh168) [inline torch/ao/nn/intrinsic/quantized/modules/conv_add.py:67]: nit: `super(...)` -> `super()`

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/nn/intrinsic/quantized/modules/conv_add.py (hunk 1 above)
Comment (leslie-fang-intel) [reply torch/ao/nn/intrinsic/quantized/modules/conv_add.py:67]: Thanks and changed.

File: test/quantization/core/test_quantized_module.py [hunk 2]
```diff
@@ -834,6 +839,66 @@ def test_conv2d_add(self):
...
+                padding = (pad_h, pad_w)
+                dilation = (dilation, dilation)
+                X_scale = 1.3
+                X_zero_point = 2
+                X2_scale = 1.2
+                X2_zero_point = 1
+                W_scale = [0.5]
+                W_zero_point = [0] if qengine_is_onednn() else [3]
+                Y_scale = 5.0
+                Y_zero_point = 4
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_module.py:880]: nit: please move these outside of the loop

Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_module.py]: Thanks and changed.

**Pull request closed without merging**